pydeseq2/utils.py

```diff
@@ -1520,3 +1522,73 @@ def make_MA_plot(
 
     if save_path is not None:
         plt.savefig(save_path, bbox_inches="tight")
+
+
+# Authors: Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+# Adapted from https://gist.github.com/agramfort/850437
+
+
+def lowess(
+    features: np.ndarray, targets: np.ndarray, frac: float = 2.0 / 3.0, iter: int = 3
+):
+    """Run lowess smoothing: Robust locally weighted regression.
+
+    The lowess function fits a nonparametric regression curve to a scatterplot.
+    The arrays features and targets contain an equal number of elements; each pair
+    (features[i], targets[i]) defines a data point in the scatterplot. The function
+    returns the estimated (smooth) values of targets.
+    The smoothing span is given by f. A larger value for f will result in a
+    smoother curve. The number of robustifying iterations is given by iter. The
+    function will run faster with a smaller number of iterations.
```

Comment:
Maybe specify that the features and labels array are assumed to be flat and of the same shape